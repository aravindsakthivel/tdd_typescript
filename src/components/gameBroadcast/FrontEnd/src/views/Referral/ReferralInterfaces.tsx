import { ReactNode, Dispatch, SetStateAction } from 'react';

export interface OverviewTotalOnboardedInterface {
  title: string;
  onBoardCount: string | number;
}

export interface ReferralpageDetailsInterface {
  userId: string;
  offset: number;
  count: number;
}

export interface ReferralLinkDetailsInterface {
  key?: string | number | undefined;
  urlLink: string;
  createdOn: string;
  destination: string;
  onBoarded: string;
}

interface GeneratedReferralInterface {
  generateReferralLink: {
    destination: string;
    referralUrl: string;
  };
}

export interface ReferralLinkModalInterface extends GeneratedReferralInterface {
  isFailure: boolean;
  isModalOpen: boolean;
  closeModal: () => void;
  holdClose: boolean;
}

export interface ReferralLinkInterface {
  referralLinkDetails: ReferralLinkDetailsInterface[];
}

export interface ReferralRowTemplateInterface {
  children: ReactNode;
  title: string;
  last: boolean;
}

export interface SelectOptionInterface {
  name: string;
  value: string;
}

// the following interface was to test the select
export interface DestinationInterface {
  destination?: 'mpl_pro' | 'mpl_qa' | 'mpl_dev' | string | undefined; // currently no destination info present using general values
}

export interface CreateReferralQueryInterface extends DestinationInterface {
  loading?: boolean;
  error?: string | undefined;
  isModalOpen: boolean;
  setIsModalOpen: Dispatch<SetStateAction<boolean>>;
  data: GeneratedReferralInterface;
}
