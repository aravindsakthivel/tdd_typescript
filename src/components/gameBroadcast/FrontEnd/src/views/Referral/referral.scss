.general-background-border {
  background: #391958 !important;
  border-bottom: 1px solid rgba(255, 255, 255, 0.3) !important;
}

.referral-link-modal .ant-modal-content,
.referral-link-modal .ant-modal-header,
.referral-link-modal .ant-modal-body,
.referral-link-modal .ant-modal-footer,
.referral-link-destination-option,
.referral-link-form .ant-select-selector {
  background-color: #492d66 !important;
  border: 1px solid rgba(255, 255, 255, 0.1) !important;
}

.general-background-border .ant-typography,
.referral-link-destination-option .ant-select-item-option-content,
.referral-link-modal .ant-modal-header .ant-modal-title,
.referral-link-modal .ant-modal-body .ant-typography,
.referral-link-modal .ant-modal-content .anticon,
.referral-link-form .ant-select-selection-item,
.referral-link-form .ant-select-arrow {
  color: #ffffff;
}

.ant-select-dropdown {
  padding: 0 !important;
}

.no-border-last {
  border: 0 !important;
}

.referral-link-modal .ant-typography-copy {
  float: right;
}

.referral-link-modal .ant-modal-close-icon {
  margin-top: 17px;
}

.referralLinkCover > div {
  margin-bottom: 0 !important;
}

.referralLinkCover > div > div {
  margin-top: 4px;
}

.referralLinkCover {
  width: 100%;
  background-color: #2d0f4d;
  padding: 5px;
}

.referral-row-template-class {
  padding: 16px 0px 16px 24px !important;
}

.button-common-class {
  background: #e91051 !important;
  font-size: 16px !important;
  line-height: 20px !important;
  font-weight: 500 !important;
  border: 0 !important;
  color: #ffffff !important;
}

.generate-link-button {
  padding: 8px 16px !important;
}

.referral-info-card .ant-typography-secondary {
  font-style: normal !important;
  font-weight: normal !important;
  font-size: 12px !important;
  line-height: 20px !important;
  opacity: 75% !important;
}

.referral-link-table .ant-table-cell {
  background-color: #391958 !important;
  font-weight: normal;
  font-size: 14px;
  line-height: 20px;
  border-bottom: 1px solid rgba(255, 255, 255, 0.3);
  color: #ffffff;
}

.referral-link-table .ant-pagination-item,
.referral-link-table .ant-pagination-item-link {
  border-color: #917fa2 !important;
}

.referral-link-table .ant-pagination-item > a,
.referral-link-table .ant-pagination-item-link {
  background-color: #917fa2 !important;
  color: #ffffff !important;
}

.referral-link-table .ant-pagination-item-active > a {
  color: black !important;
  background-color: #ffffff !important;
}
