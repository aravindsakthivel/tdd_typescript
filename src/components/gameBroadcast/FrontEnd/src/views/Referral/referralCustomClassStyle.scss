.manage-user-page-class {
  padding: 40px 48px 24px 24px !important;
}

.referral-row-template-class {
  padding: 16px 0px 16px 24px !important;
}

.add-user-button-class {
  width: 172px;
  height: 48px;
}

.button-common-class {
  background: #e91051 !important;
  font-size: 16px !important;
  line-height: 20px !important;
  font-weight: 500 !important;
  border: 0 !important;
  color: #ffffff !important;
}

.generate-link-button {
  padding: 8px 16px !important;
}

