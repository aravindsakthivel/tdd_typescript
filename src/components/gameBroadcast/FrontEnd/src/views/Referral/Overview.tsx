import React, { ReactElement } from 'react';
import ReferralRowTemplate from './ReferralRowTemplate';
import TotalOnboarded from './TotalOnboarded';
import { OverviewTotalOnboardedInterface } from './ReferralInterfaces';

function Overview({
  title = 'User Onboarded this month',
  onBoardCount = 324,
}: OverviewTotalOnboardedInterface): ReactElement {
  return (
    <ReferralRowTemplate title="Overview" last={false}>
      <TotalOnboarded title={title} onBoardCount={onBoardCount} />
    </ReferralRowTemplate>
  );
}

export default Overview;
