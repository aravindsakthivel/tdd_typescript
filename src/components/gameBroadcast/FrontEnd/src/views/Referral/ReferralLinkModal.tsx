import React, { ReactElement } from 'react';
import { Modal, Typography } from 'antd';
import Spinner from '../Common/Spinner';
import { ReferralLinkModalInterface } from './ReferralInterfaces';

const { Title, Paragraph } = Typography;

function ReferralLinkModal({
  isModalOpen,
  closeModal,
  holdClose,
  isFailure = false,
  generateReferralLink,
}: ReferralLinkModalInterface): ReactElement {
  return (
    <>
      <Modal
        visible={isModalOpen}
        title="New Link"
        footer={null}
        onCancel={holdClose ? () => {} : closeModal}
        maskStyle={{ backgroundColor: '#5A4074', opacity: 0.95 }}
        data-testid="referral-link-modal-component"
        className="referral-link-modal">
        {holdClose ? (
          <Spinner />
        ) : (
          <>
            {isFailure ? (
              <Paragraph>
                Something went wrong &nbsp; Please try again or check link table
              </Paragraph>
            ) : (
              generateReferralLink !== undefined && (
                <>
                  <Title
                    level={4}
                    data-testid="referral-modal-destination-title">
                    Destination:{' '}
                    {generateReferralLink.destination
                      .split('_')
                      .join(' ')
                      .toUpperCase() || ''}
                  </Title>
                  <div className="referralLinkCover">
                    <Paragraph copyable>
                      {generateReferralLink.referralUrl || ''}
                    </Paragraph>
                  </div>
                </>
              )
            )}
          </>
        )}
      </Modal>
    </>
  );
}

export default ReferralLinkModal;
