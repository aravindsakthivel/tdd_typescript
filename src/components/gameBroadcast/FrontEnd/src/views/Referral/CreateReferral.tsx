import React, { useState, ReactElement } from 'react';
import { useMutation } from '@apollo/client';
import { Select, Button, Form } from 'antd';
import ReferralRowTemplate from './ReferralRowTemplate';
import ReferralLinkModal from './ReferralLinkModal';
import { SuccessMessage, ErrorMessage } from '../Common/SequentialMessage';
import { GENERATE_REFERRAL_LINK } from '../../queries';
import {
  SelectOptionInterface,
  DestinationInterface,
} from './ReferralInterfaces';

const { Option } = Select;

function CreateReferral({
  destination: destinationTest = '',
}: DestinationInterface): ReactElement {
  const [generateReferrals, { data, loading, error }] = useMutation(
    GENERATE_REFERRAL_LINK,
    {
      notifyOnNetworkStatusChange: true,
    },
  );
  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
  const [destination, setDestination] = useState<string>(destinationTest);
  // currently no destination info present using general values
  const optionValues: SelectOptionInterface[] = [
    {
      name: 'MPL PRO',
      value: 'mpl_pro',
    },
    {
      name: 'MPL QA',
      value: 'mpl_qa',
    },
    {
      name: 'MPL DEV',
      value: 'mpl_dev',
    },
  ];

  const onChangeDestination = (option: string) => {
    setDestination(option);
  };

  const onSubmit = () => {
    generateReferrals({
      variables: {
        generateReferralLinkData: { userId: 'skdfjlksj589', destination },
      },
    })
      .then(() => {
        SuccessMessage('Referral Link Generated Successfully', 3);
      })
      .catch(() => {
        ErrorMessage('Something went wrong try again', 3);
      });
    setIsModalOpen(!isModalOpen);
  };

  return (
    <>
      <ReferralRowTemplate title="Create a custom link" last={false}>
        <Form
          layout="inline"
          size="large"
          onFinish={onSubmit}
          data-testid="generate-link-form"
          initialValues={
            destinationTest !== '' ? { destination: destinationTest } : {}
          }
          className="referral-link-form">
          <Form.Item
            name="destination"
            rules={[
              { required: true, message: 'Please select a Destination' },
            ]}>
            <Select
              style={{ width: 350, borderRadius: 10 }}
              placeholder="Select a Destination"
              onChange={onChangeDestination}
              data-testid="select-destination-option">
              {optionValues.length > 0 &&
                optionValues.map(({ name, value }, ind) => (
                  <Option
                    value={value}
                    // eslint-disable-next-line react/no-array-index-key
                    key={ind}
                    className="referral-link-destination-option">
                    {name}
                  </Option>
                ))}
            </Select>
          </Form.Item>
          <Form.Item>
            <Button
              type="primary"
              htmlType="submit"
              className="button-common-class generate-link-button"
              data-testid="generate-link-button">
              Generate Link
            </Button>
          </Form.Item>
        </Form>
      </ReferralRowTemplate>
      {isModalOpen && (
        <ReferralLinkModal
          isModalOpen={isModalOpen}
          closeModal={() => setIsModalOpen(false)}
          holdClose={!!loading}
          isFailure={!(error === undefined)}
          generateReferralLink={
            data !== undefined ? data.generateReferralLink : undefined
          }
        />
      )}
    </>
  );
}

export default CreateReferral;
