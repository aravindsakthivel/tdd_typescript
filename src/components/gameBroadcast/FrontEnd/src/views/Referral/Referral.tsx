import React, { FC, ReactElement } from 'react';
import { useQuery } from '@apollo/client';
import { Row, Typography } from 'antd';
import { GET_REFERRAL_PAGE_DETAILS } from '../../queries';
import Spinner from '../Common/Spinner';
import Overview from './Overview';
import ReferrralLinkTable from './ReferralLinkTable';
import CreateReferral from './CreateReferral';
import {
  ReferralLinkDetailsInterface,
  ReferralpageDetailsInterface,
} from './ReferralInterfaces';
import './referral.scss';

const { Title } = Typography;

function ReferralPageDetails({
  userId,
  offset,
  count,
}: ReferralpageDetailsInterface): ReactElement | null {
  const { loading, error, data, networkStatus } = useQuery(
    GET_REFERRAL_PAGE_DETAILS,
    {
      variables: { getReferralLinksData: { userId, offset, count } },
      notifyOnNetworkStatusChange: true,
    },
  );

  if (networkStatus === 4) return <p>Re-fetching!</p>;
  if (loading) return <Spinner />;
  if (error) return <div>`Error!: ${JSON.stringify(error)}`</div>;
  const {
    getReferralLinks: { referralLinkDetails },
  } = data;

  let onBoardCount = 0;
  const referralLinks: ReferralLinkDetailsInterface[] = [];

  referralLinkDetails.forEach(
    (referrals: ReferralLinkDetailsInterface, ind: number) => {
      onBoardCount += Number(referrals.onBoarded);
      referralLinks.push({ ...referrals, key: ind });
    },
  );

  return (
    <>
      <CreateReferral />
      <Overview title="User Onboarded this month" onBoardCount={onBoardCount} />
      <ReferrralLinkTable referralLinkDetails={referralLinks} />
    </>
  );
}

const Referral: FC = () => {
  return (
    <>
      <Row className="general-background-border">
        <Title
          level={2}
          style={{ padding: '40px 0px 24px 24px', marginBottom: 0 }}
          data-testid="referral-page-title">
          Referral
        </Title>
      </Row>
      <ReferralPageDetails userId="sdfdsg" offset={0} count={0} />
    </>
  );
};

export default Referral;
