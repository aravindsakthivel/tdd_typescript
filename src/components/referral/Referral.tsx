import { FC } from 'react';
import { Row, Typography } from 'antd';
import { CreateReferral } from './CreateReferral';
import { Overview } from './Overview';
import { ReferrralLinkTable } from './ReferrralLinkTable';
import './referral.css';

const { Title } = Typography;

const Referral: FC = () => {
  return (
    <>
      <Row className="general-background-border">
        <Title
          level={2}
          style={{ padding: '40px 0px 24px 24px', marginBottom: 0 }}
          data-testid="referral-page-title"
        >
          Referral
        </Title>
      </Row>
      <CreateReferral />
      <Overview title="User Onboarded this month" onBoardCount={324} />
      <ReferrralLinkTable />
    </>
  );
};

export { Referral };
