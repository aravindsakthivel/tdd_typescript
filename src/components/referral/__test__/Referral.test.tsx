import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { mockConfig } from '../../mock';
import { Referral } from '../Referral';

afterEach(cleanup);
beforeEach(() => {
  mockConfig();
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Referral />, div);
});

it('The component consist of title', () => {
  const { getByTestId } = render(<Referral />);
  expect(getByTestId('referral-page-title')).toHaveTextContent('Referral');
});

it('matches snapshot', () => {
  const tree = renderer.create(<Referral />).toJSON();
  expect(tree).toMatchSnapshot();
});
