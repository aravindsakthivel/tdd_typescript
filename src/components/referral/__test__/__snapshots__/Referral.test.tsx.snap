// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div
    className="ant-row general-background-border"
    style={Object {}}
  >
    <h2
      className="ant-typography"
      data-testid="referral-page-title"
      style={
        Object {
          "WebkitLineClamp": undefined,
          "marginBottom": 0,
          "padding": "40px 0px 24px 24px",
        }
      }
    >
      Referral
    </h2>
  </div>,
  <div
    className="ant-row referral-row-template-class general-background-border false"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-24"
      style={Object {}}
    >
      <h5
        className="ant-typography"
        data-testid="referral-row-title"
        style={
          Object {
            "WebkitLineClamp": undefined,
          }
        }
      >
        Create a custom link
      </h5>
    </div>
    <div
      className="ant-col ant-col-23"
      data-testid="referral-row-children"
      style={Object {}}
    >
      <form
        className="ant-form ant-form-inline ant-form-large referral-link-form"
        data-testid="generate-link-form"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div
          className="ant-row ant-form-item"
          style={Object {}}
        >
          <div
            className="ant-col ant-form-item-control"
            style={Object {}}
          >
            <div
              className="ant-form-item-control-input"
            >
              <div
                className="ant-form-item-control-input-content"
              >
                <div
                  className="ant-select ant-select-lg ant-select-single ant-select-show-arrow"
                  data-testid="select-destination-option"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  style={
                    Object {
                      "borderRadius": 10,
                      "width": 350,
                    }
                  }
                >
                  <div
                    className="ant-select-selector"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                  >
                    <span
                      className="ant-select-selection-search"
                    >
                      <input
                        aria-activedescendant="destination_list_0"
                        aria-autocomplete="list"
                        aria-controls="destination_list"
                        aria-haspopup="listbox"
                        aria-owns="destination_list"
                        autoComplete="off"
                        className="ant-select-selection-search-input"
                        id="destination"
                        onChange={[Function]}
                        onCompositionEnd={[Function]}
                        onCompositionStart={[Function]}
                        onKeyDown={[Function]}
                        onMouseDown={[Function]}
                        onPaste={[Function]}
                        readOnly={true}
                        role="combobox"
                        style={
                          Object {
                            "opacity": 0,
                          }
                        }
                        type="search"
                        unselectable="on"
                        value=""
                      />
                    </span>
                    <span
                      className="ant-select-selection-placeholder"
                    >
                      Select a Destination
                    </span>
                  </div>
                  <span
                    aria-hidden={true}
                    className="ant-select-arrow"
                    onMouseDown={[Function]}
                    style={
                      Object {
                        "WebkitUserSelect": "none",
                        "userSelect": "none",
                      }
                    }
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      className="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-row ant-form-item"
          style={Object {}}
        >
          <div
            className="ant-col ant-form-item-control"
            style={Object {}}
          >
            <div
              className="ant-form-item-control-input"
            >
              <div
                className="ant-form-item-control-input-content"
              >
                <button
                  className="ant-btn ant-btn-primary ant-btn-lg button-common-class generate-link-button"
                  data-testid="generate-link-button"
                  onClick={[Function]}
                  type="submit"
                >
                  <span>
                    Generate Link
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>,
  <div
    className="ant-row referral-row-template-class general-background-border false"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-24"
      style={Object {}}
    >
      <h5
        className="ant-typography"
        data-testid="referral-row-title"
        style={
          Object {
            "WebkitLineClamp": undefined,
          }
        }
      >
        Overview
      </h5>
    </div>
    <div
      className="ant-col ant-col-23"
      data-testid="referral-row-children"
      style={Object {}}
    >
      <div
        className="ant-card ant-card-bordered referral-info-card"
        style={
          Object {
            "background": "#492D66",
            "border": 0,
            "height": 108,
            "width": 218,
          }
        }
      >
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <span
            className="ant-typography ant-typography-secondary"
            data-testid="info-card-title"
            style={
              Object {
                "WebkitLineClamp": undefined,
                "color": "#ffffff",
              }
            }
          >
            User Onboarded this month
          </span>
          <h3
            className="ant-typography"
            data-testid="info-card-onboard-count"
            style={
              Object {
                "WebkitLineClamp": undefined,
                "marginTop": "5px",
              }
            }
          >
            324
          </h3>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="ant-row referral-row-template-class general-background-border no-border-last"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-24"
      style={Object {}}
    >
      <h5
        className="ant-typography"
        data-testid="referral-row-title"
        style={
          Object {
            "WebkitLineClamp": undefined,
          }
        }
      >
        History
      </h5>
    </div>
    <div
      className="ant-col ant-col-23"
      data-testid="referral-row-children"
      style={Object {}}
    >
      <div
        className="ant-table-wrapper referral-link-table"
        style={
          Object {
            "backgroundColor": "#391958",
          }
        }
      >
        <div
          className="ant-spin-nested-loading"
        >
          <div
            className="ant-spin-container"
          >
            <div
              className="ant-table"
            >
              <div
                className="ant-table-container"
              >
                <div
                  className="ant-table-content"
                  onScroll={[Function]}
                  style={Object {}}
                >
                  <table
                    style={
                      Object {
                        "tableLayout": "auto",
                      }
                    }
                  >
                    <colgroup />
                    <thead
                      className="ant-table-thead"
                    >
                      <tr>
                        <th
                          className="ant-table-cell"
                          colSpan={null}
                          rowSpan={null}
                          style={Object {}}
                        >
                          URL / Link
                        </th>
                        <th
                          className="ant-table-cell"
                          colSpan={null}
                          rowSpan={null}
                          style={Object {}}
                        >
                          Created On
                        </th>
                        <th
                          className="ant-table-cell"
                          colSpan={null}
                          rowSpan={null}
                          style={Object {}}
                        >
                          Destination
                        </th>
                        <th
                          className="ant-table-cell"
                          colSpan={null}
                          rowSpan={null}
                          style={Object {}}
                        >
                          Onboarded
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      className="ant-table-tbody"
                    >
                      <tr
                        className="ant-table-row ant-table-row-level-0"
                        data-row-key="1"
                        onClick={[Function]}
                        style={Object {}}
                      >
                        <td
                          className="ant-table-cell"
                          colSpan={null}
                          rowSpan={null}
                          style={Object {}}
                        >
                          https://mpl.live/mycustomcodeforamqzingoffers 
                        </td>
                        <td
                          className="ant-table-cell"
                          colSpan={null}
                          rowSpan={null}
                          style={Object {}}
                        >
                          12:01pm, 12 Mar 2020
                        </td>
                        <td
                          className="ant-table-cell"
                          colSpan={null}
                          rowSpan={null}
                          style={Object {}}
                        >
                          1 days
                        </td>
                        <td
                          className="ant-table-cell"
                          colSpan={null}
                          rowSpan={null}
                          style={Object {}}
                        >
                          12
                        </td>
                      </tr>
                      <tr
                        className="ant-table-row ant-table-row-level-0"
                        data-row-key="2"
                        onClick={[Function]}
                        style={Object {}}
                      >
                        <td
                          className="ant-table-cell"
                          colSpan={null}
                          rowSpan={null}
                          style={Object {}}
                        >
                          https://mpl.live/mycustomcodeforamqzingoffers 
                        </td>
                        <td
                          className="ant-table-cell"
                          colSpan={null}
                          rowSpan={null}
                          style={Object {}}
                        >
                          12:01pm, 12 Mar 2020
                        </td>
                        <td
                          className="ant-table-cell"
                          colSpan={null}
                          rowSpan={null}
                          style={Object {}}
                        >
                          3 days
                        </td>
                        <td
                          className="ant-table-cell"
                          colSpan={null}
                          rowSpan={null}
                          style={Object {}}
                        >
                          33
                        </td>
                      </tr>
                      <tr
                        className="ant-table-row ant-table-row-level-0"
                        data-row-key="3"
                        onClick={[Function]}
                        style={Object {}}
                      >
                        <td
                          className="ant-table-cell"
                          colSpan={null}
                          rowSpan={null}
                          style={Object {}}
                        >
                          https://mpl.live/mycustomcodeforamqzingoffers 
                        </td>
                        <td
                          className="ant-table-cell"
                          colSpan={null}
                          rowSpan={null}
                          style={Object {}}
                        >
                          12:01pm, 12 Mar 2020
                        </td>
                        <td
                          className="ant-table-cell"
                          colSpan={null}
                          rowSpan={null}
                          style={Object {}}
                        >
                          7 days
                        </td>
                        <td
                          className="ant-table-cell"
                          colSpan={null}
                          rowSpan={null}
                          style={Object {}}
                        >
                          325
                        </td>
                      </tr>
                      <tr
                        className="ant-table-row ant-table-row-level-0"
                        data-row-key="4"
                        onClick={[Function]}
                        style={Object {}}
                      >
                        <td
                          className="ant-table-cell"
                          colSpan={null}
                          rowSpan={null}
                          style={Object {}}
                        >
                          https://mpl.live/mycustomcodeforamqzingoffers 
                        </td>
                        <td
                          className="ant-table-cell"
                          colSpan={null}
                          rowSpan={null}
                          style={Object {}}
                        >
                          12:01pm, 12 Mar 2020
                        </td>
                        <td
                          className="ant-table-cell"
                          colSpan={null}
                          rowSpan={null}
                          style={Object {}}
                        >
                          8 days
                        </td>
                        <td
                          className="ant-table-cell"
                          colSpan={null}
                          rowSpan={null}
                          style={Object {}}
                        >
                          5466
                        </td>
                      </tr>
                      <tr
                        className="ant-table-row ant-table-row-level-0"
                        data-row-key="5"
                        onClick={[Function]}
                        style={Object {}}
                      >
                        <td
                          className="ant-table-cell"
                          colSpan={null}
                          rowSpan={null}
                          style={Object {}}
                        >
                          https://mpl.live/mycustomcodeforamqzingoffers 
                        </td>
                        <td
                          className="ant-table-cell"
                          colSpan={null}
                          rowSpan={null}
                          style={Object {}}
                        >
                          12:01pm, 12 Mar 2020
                        </td>
                        <td
                          className="ant-table-cell"
                          colSpan={null}
                          rowSpan={null}
                          style={Object {}}
                        >
                          7 days
                        </td>
                        <td
                          className="ant-table-cell"
                          colSpan={null}
                          rowSpan={null}
                          style={Object {}}
                        >
                          325
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
